frontend/src/scenes/settings/project/HostedReverseProxy.tsx

```diff
@@ -87,24 +86,28 @@ export function Proxy(): JSX.Element {
         },
     ]
 
-    return (
-        <div className="space-y-2">
-            <LemonTable
-                loading={proxyRecords.length === 0 && proxyRecordsLoading}
-                columns={columns}
-                dataSource={proxyRecords}
-                expandable={{
-                    expandedRowRender: (record) => <ExpandedRow record={record} />,
-                }}
-            />
-            {formState === 'collapsed' ? (
-                <LemonButton onClick={showForm} type="secondary" icon={<IconPlus />}>
-                    Add domain
-                </LemonButton>
-            ) : (
-                <CreateRecordForm />
-            )}
-        </div>
+    return isCloudOrDev ? (
+        <PayGateMini feature={AvailableFeature.HOSTED_REVERSE_PROXY}>
+            <div className="space-y-2">
+                <LemonTable
+                    loading={proxyRecords.length === 0 && proxyRecordsLoading}
+                    columns={columns}
+                    dataSource={proxyRecords}
+                    expandable={{
+                        expandedRowRender: (record) => <ExpandedRow record={record} />,
+                    }}
+                />
+                {formState === 'collapsed' ? (
+                    <LemonButton onClick={showForm} type="secondary" icon={<IconPlus />}>
+                        Add domain
+                    </LemonButton>
+                ) : (
+                    <CreateRecordForm />
+                )}
+            </div>
+        </PayGateMini>
+    ) : (
+        <LemonBanner type="warning">Using a reverse proxy only works in PostHog Cloud</LemonBanner>
```

Comment:
They paygate actually has CTAs for posthog cloud, so I think this check is unnecessary.